src/FileRouter.php

```diff
@@ -155,61 +155,57 @@ private function parseRequestPath(ServerRequestInterface $request): Generator
             }
         }
 
-        $possibleAction = null;
-
         if ($path === '/') {
             yield [
-                $this->cleanClassname(
-                    $this->namespace . '\\' . $this->baseControllerDirectory . '\\' . $this->defaultControllerName . $this->classPostfix
-                ),
-                $possibleAction,
+                $this->makeClassName($this->defaultControllerName),
+                null,
             ];
             return;
         }
 
         $controllerName = preg_replace_callback(
-            '#(/.)#u',
-            static fn(array $matches) => mb_strtoupper($matches[1]),
+            '#/.#u',
+            static fn(array $matches) => mb_strtoupper($matches[0]),
             $path,
         );
 
-        if (!preg_match('#^(.*?)/([^/]+)/?$#', $controllerName, $matches)) {
+        if (!preg_match('#^/?(.*?)/([^/]+)/?$#', $controllerName, $matches)) {
             return;
         }
 
-        $directoryPath = $matches[1];
-        $controllerName = $matches[2];
+        [$_, $directoryPath, $controllerName] = $matches;
 
         yield [
-            $this->cleanClassname(
-                $this->namespace . '\\' . $this->baseControllerDirectory . '\\' . $directoryPath . '\\' . $controllerName . $this->classPostfix
-            ),
-            $possibleAction,
+            $this->makeClassName($controllerName, $directoryPath),
+            null,
         ];
 
-        if (preg_match('#^(.*?)/([^/]+)/?$#', $directoryPath, $matches)) {
-            $possibleAction = strtolower($controllerName);
-            $directoryPath = $matches[1];
-            $controllerName = $matches[2];
+        if ($directoryPath === '') {
+            yield [
+                $this->makeClassName($this->defaultControllerName, $controllerName),
+                null,
+            ];
         } else {
-            $directoryPath = $controllerName;
-            $controllerName = $this->defaultControllerName;
+            yield [
+                $this->makeClassName($directoryPath),
+                strtolower($controllerName),
+            ];
         }
-
-        yield [
-            $this->cleanClassname(
-                $this->namespace . '\\' . $this->baseControllerDirectory . '\\' . $directoryPath . '\\' . $controllerName . $this->classPostfix
-            ),
-            $possibleAction,
-        ];
     }
 
-    private function cleanClassname(string $className): string
+    private function makeClassName(string $controllerName, string $directoryPath = ''): string
     {
-        return str_replace(
```

Comment:
As far as I remember there were some Windows cases where such operations are necessary to make the router work correctly

Reply:
Seems, path always use `/`. See https://github.com/php-fig/http-message/blob/402d35bcb92c70c026d1a6a9883f06b2ead23d71/src/UriInterface.php#L135